Formula/adguard-home.rb

```diff
@@ -0,0 +1,113 @@
+class AdguardHome < Formula
+  desc "Network-wide ads & trackers blocking DNS server"
+  homepage "https://adguard.com/adguard-home.html"
+  url "https://github.com/AdguardTeam/AdGuardHome.git",
+      :tag      => "v0.98.1",
+      :revision => "64d40bdc47081ed80e597e61e5c96953c1e35c30"
+  head "https://github.com/AdguardTeam/AdGuardHome.git", :using => :git
+
+  depends_on "go" => :build
+  depends_on "node" => :build
+
+  resource "packr" do
+    url "https://github.com/gobuffalo/packr/archive/v2.7.1.tar.gz"
+    sha256 "842bc86dfb34c1ff8d1cc1c12fb46cf30ad279eec17ecef8b122c151cfc16b85"
+  end
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["CGO_ENABLED"] = "0"
+
+    ENV.append_path "PATH", HOMEBREW_PREFIX/"bin"
+
+    dir = buildpath/"src/github.com/AdguardTeam/AdGuardHome"
+    dir.install buildpath.children
+
+    resource("packr").stage { system "go", "install", "./packr" }
+
+    cd dir do
+      system "make"
```

Comment:
Can you double check if this approach (compared to go_resource) will avoid downloading packr again from the latest master?

Reply:
Makefile target will most likely try to download the `packr`, see [Makefile#L24](https://github.com/AdguardTeam/AdGuardHome/blob/master/Makefile#L24). I'm not familiar enough with how go resolves packages. I can try removing the resource and see if Makefile works.

I was unable to use `go_resource` properly though. Brew downloaded the package, but `packr` binary wasn't available from Makefile. I probably have to manually alter the PATH env?

Reply:
`go_resource` (or `resource` in this case) exist basically to pre-download and allow caching things in the right place. If you put them in the same place (with symlinks if necessary) that the `Makefile` `go get` line will that should be sufficient to avoid the re-download.

Reply:
If that's too tricky to figure out: let me know and I'll jump in and give it a go.

Reply:
@MikeMcQuaid `Makefile` just downloads the latest version of all the required packages to `buildpath`. It's not necessary to declare a separate resource now when the formula uses makefile. There's a dozen of other go packages downloaded. I don't see benefits of caching only `packr` nor in specifying all the dependencies as resources. I've tested the build and it works fine.

I'm going to remove the resource part and push changes.

Reply:
> I don't see benefits of caching only `packr` nor in specifying all the dependencies as resources?

There are a few benefits/requirements:
- the formula doesn't break if the upstream `packr` `master` branch breaks (this has happened to other formulae before and is very hard to debug and a pain to fix after the fact
- there is no caching as all downloaded `go` packages are removed every time

Ideally the upstream would make use of something like `dep` but if we don't have that: `go_resource` should be the easiest way to ensure these two problems are avoided.

Reply:
OK. Do you require to use `go_resource` for **every** dependency that this project is using and downloading? Same reasoning could be applied for all dependencies that are downloaded using npm.

Reply:
The main difference is that `npm` is not pulling everything from the tip of the `master` branch. The latest `go` module/vendoring system avoids this problem.

Reply:
Things are a bit mad the next few days with me with work but I'm happy to help more with this if that would be useful (i.e. add some of the `go_resource`s for you).